broker/src/test/java/io/camunda/zeebe/broker/system/partitions/impl/steps/StreamProcessorTransitionStepTest.java

```diff
@@ -0,0 +1,152 @@
+
+  private StreamProcessorTransitionStep step;
+
+  @BeforeEach
+  void setup() {
+    transitionContext.setLogStream(mock(LogStream.class));
+    transitionContext.setComponentHealthMonitor(mock(HealthMonitor.class));
+
+    doReturn(streamProcessor).when(streamProcessorBuilder).build();
+
+    when(streamProcessor.openAsync(anyBoolean())).thenReturn(TestActorFuture.completedFuture(null));
+    when(streamProcessor.closeAsync()).thenReturn(TestActorFuture.completedFuture(null));
+
+    step = new StreamProcessorTransitionStep(streamProcessorBuilder);
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideTransitionsThatShouldReInstallStreamProcessor")
+  void shouldCloseExistingStreamProcessor(final Role currentRole, final Role targetRole) {
+    // given
+    if (currentRole != null) {
+      transitionTo(currentRole);
+    }
+
+    // when
+    step.prepareTransition(transitionContext, 1, targetRole).join();
+
+    // then
+    assertThat(transitionContext.getStreamProcessor()).isNull();
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideTransitionsThatShouldReInstallStreamProcessor")
+  void shouldReInstallStreamProcessor(final Role currentRole, final Role targetRole) {
+    // given
+    if (currentRole != null) {
+      transitionTo(currentRole);
+    }
+    Mockito.clearInvocations(streamProcessor);
+
+    // when
+    transitionTo(targetRole);
+
+    // then
+    assertThat(transitionContext.getStreamProcessor()).isNotNull();
+    verify(streamProcessor).openAsync(anyBoolean());
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideTransitionsThatShouldDoNothing")
+  void shouldNotCloseExistingStreamProcessor(final Role currentRole, final Role targetRole) {
+    // given
+    transitionTo(currentRole);
+
+    // when
+    step.prepareTransition(transitionContext, 1, targetRole).join();
+
+    // then
+    assertThat(transitionContext.getStreamProcessor()).isNotNull();
+    verify(streamProcessor, never()).closeAsync();
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideTransitionsThatShouldDoNothing")
+  void shouldNotReInstallStreamProcessor(final Role currentRole, final Role targetRole) {
+    // given
+    transitionTo(currentRole);
+
+    // when
+    transitionTo(targetRole);
+
+    // then
+    assertThat(transitionContext.getStreamProcessor()).isNotNull();
+    verify(streamProcessor).openAsync(anyBoolean());
+    verify(streamProcessor, never()).closeAsync();
+  }
+
+  @ParameterizedTest
+  @EnumSource(
+      value = Role.class,
+      names = {"FOLLOWER", "LEADER", "CANDIDATE"})
+  void shouldCloseStreamProcessorWhenTransitioningToInactive(final Role currentRole) {
+    // given
+    transitionTo(currentRole);
+
+    // when
+    transitionTo(Role.INACTIVE);
+
+    // then
+    assertThat(transitionContext.getStreamProcessor()).isNull();
+    verify(streamProcessor).closeAsync();
+  }
+
+  private static Stream<Arguments> provideTransitionsThatShouldReInstallStreamProcessor() {
+    return Stream.of(
```

Comment:
See comment on other test regarding transitions not possible in Raft.

Also missing in this test (from Raft state machine):
* null -> INACTIVE
* FOLLOWER -> INACTIVE
* LEADER -> INACTIVE

Reply:
There is a specific test for transition to Inactive `shouldCloseStreamProcessorWhenTransitioningToInactive`.